source/config/Config.cpp

```diff
@@ -539,6 +615,32 @@ string PlainConfig::LogConfig::ParseDeviceClientLogType(string value)
     }
 }
 
+string PlainConfig::LogConfig::StringifyDeviceClientLogLevel(int level) const
+{
+
+    switch (level)
+    {
+        case 0:
+            return "error";
+            break;
+        case 1:
+            return "warn";
+            break;
+        case 2:
+            return "info";
+            break;
+    }
+    // this is the defualt case for log level
```

Comment:
* Add an explicit case for Debug to be consistent with out-of-range handling in the other functions that convert log level.
* Replace the magic numbers with the named constants. This requires you to use a `static_cast` to convert `int` to `enum class LogLevel`.

```
switch (static_cast<DeviceClient::Logging::LogLevel>(level)):
{
    case DeviceClient::Logging::LogLevel::ERROR:
        return "ERROR";
    ...
    case DeviceClient::Logging::LogLevel::DEBUG:
        return "DEBUG";
}
throw std::invalid_argument(...);
```